### PR TITLE
refactor(box libfuncs): use BlockExt trait

### PR DESCRIPTION
**Pull request by FeliGR**
## Checklist
- [ lambdaclass/cairo_native/issues/521 ] Linked to Github Issue
- [ ] Unit tests added
- [ ] Integration tests added.
- [ ] This change requires new documentation.
  - [ ] Documentation has been added/updated.

### CODE REVIEW

File: src/libfuncs/box.rs [hunk 1]
```diff
@@ -73,18 +69,7 @@ pub fn build_into_box<'ctx, 'this>(
...
-            IntegerAttribute::new(
-                IntegerType::new(context, 64).into(),
-                inner_layout.pad_to_align().size().try_into()?,
-            )
-            .into(),
-            location,
-        ))
-        .result(0)?
-        .into();
+    let value_len = entry.const_int(context, location, inner_layout.size() as i64, 64)?;
```
[the remark this answers is not in the document]
Comment (FeliGR) [reply src/libfuncs/box.rs]: @edg-l  fixed.

File: src/libfuncs/box.rs [hunk 2]
```diff
@@ -155,35 +137,14 @@ pub fn build_unbox<'ctx, 'this>(
...
-                        .align(Some(IntegerAttribute::new(
-                            IntegerType::new(context, 64).into(),
-                            inner_layout.align() as i64,
-                        )))
-                        .elem_type(Some(TypeAttribute::new(inner_ty))),
-                ))
-                .result(0)?
-                .into();
+            let value_len = entry.const_int(context, location, inner_layout.size() as i64, 64)?;
+            let stack_ptr = entry.alloca(
```
[the remark this answers is not in the document]
Comment (FeliGR) [reply src/libfuncs/box.rs]: @edg-l  fixed.

**Review by edg-l — APPROVED** [review]

File: src/libfuncs/box.rs [hunk 3]
```diff
@@ -73,18 +69,7 @@ pub fn build_into_box<'ctx, 'this>(
...
-            IntegerAttribute::new(
-                IntegerType::new(context, 64).into(),
-                inner_layout.pad_to_align().size().try_into()?,
-            )
-            .into(),
-            location,
-        ))
-        .result(0)?
-        .into();
+    let value_len = entry.const_int(context, location, inner_layout.size(), 64)?;
```
Comment (azteca1998) [inline src/libfuncs/box.rs:72]: I think you forgot the `.pad_to_align()`.

Comment (FeliGR) [reply src/libfuncs/box.rs]: @azteca1998 fixed.

**Review by azteca1998 — DISMISSED** [review]

**Review by edg-l — DISMISSED** [review]

**Review by juanbono — APPROVED** [review]

**Comment by juanbono** [comment]
Please fix conflict @FeliGR

**Review by juanbono — APPROVED** [review]

**Review by juanbono — APPROVED** [review]

File: src/libfuncs/box.rs [hunk 4]
```diff
@@ -68,18 +66,7 @@ pub fn build_into_box<'ctx, 'this>(
...
-            )
-            .into(),
-            location,
-        ))
-        .result(0)?
-        .into();
+    let value_len = entry.const_int(context, location, inner_layout.pad_to_align().size(), 64)?;
 
     let ptr = entry
         .append_operation(llvm::nullptr(opaque_pointer(context), location))
```
Comment (tcoratger) [inline src/libfuncs/box.rs:72]: Here this should be refactored using `append_op_result` no?

Comment (azteca1998) [reply src/libfuncs/box.rs:72]: Yes, and the next operation too (the ReallocBindingsMeta::realloc).

**Pull request closed without merging**